packages/query-core/src/tests/hydration.test.tsx

```diff
@@ -426,4 +427,18 @@ describe('dehydration and rehydration', () => {
 
     queryClient.clear()
   })
+
+  test('should set the fetchStatus to idle in all cases when dehydrating', async () => {
+    const queryCache = new QueryCache()
+    const queryClient = createQueryClient({ queryCache })
+    await queryClient.prefetchQuery(['string'], () => fetchData('string', 10))
+
+    queryClient.refetchQueries(['string'])
+
+    const dehydrated = dehydrate(queryClient)
+    const stringified = JSON.stringify(dehydrated)
+
+    const parsed = JSON.parse(stringified) as DehydratedState
+    expect(parsed.queries[0]?.state.fetchStatus).toBe('idle')
+  })
```

Comment:
I can see that we are not awaiting `refetchQueries`, but this could still result in race conditions if CI is slow because it relies on timing (10ms sleep). I think a better way would be to have the queryFn return a Promise that we only resolve after we have called dehydrate, and maybe even expect that the `dehydrated` json contains a query in loading and fetching state?

Reply:
@TkDodo I have updated the code to ensure that we manually resolve Promise after dehydration, can you also check if it looks good? Also I am not sure how the dehydrated json would contain a query in loading and fetching state since we set the `fetchStatus` to `idle` when dehydrating.

Reply:
oh, the change is made in `dehydrate`. For some reason, I thought we apply the `fetchStatus:'idle'` overwrite when writing back to the queryCache, so when `hydrating`. Wouldn't that be better ?

Reply:
Yeah you are right, doing it while hydration makes more sense. Have updated the code and test can you check again?